solutions/usaco-764.mdx

```diff
@@ -80,6 +80,68 @@ int main(){
 }
 ```
 
+</CPPSection>
+	
+<CPPSection>
+	
+## Alternative Explanation
+	
+This problem can actually be solved using the first answer to “Badge.” In that question, the nodes whose answer was itself were the ones that were in a cycle. Otherwise, it would take a few steps to reach a cycle. If a node is in a cycle, it will never be empty. Otherwise, it will at some point be empty. You can literally just copy and paste the previous solution. Then, count how many nodes’ answer are itself. 
+	
+## Alternative Implementation
+	
+```cpp
+int n, p[INF], ans[INF];
+bool in_cycle = false;
+
+void dfs(int node) {
+   if (ans[node] != -2) {
```

Comment:
comments would be kinda nice

Reply:
Sorry about that. I'll continue to fix all the issues.